flycheck.el

```diff
@@ -6241,6 +6242,39 @@ See URL `https://github.com/kisielk/errcheck'."
     ;; packages, and can't check individual Go files.
     (and (flycheck-buffer-saved-p) (flycheck-go-package-name))))
 
+(defun flycheck-groovy-write-compiler ()
+  "Write the compilation program to temporary file so it can be executed."
+  (let* ((dir (flycheck-temp-dir-system))
+         (file (expand-file-name "compiler.groovy" dir))
+         (prog "import org.codehaus.groovy.control.*
+
+file = new File(args[0])
+unit = new CompilationUnit()
+unit.addSource(file)
+
+try {
+    unit.compile(Phases.CONVERSION)
+} catch (MultipleCompilationErrorsException e) {
+    e.errorCollector.write(new PrintWriter(System.out, true), null)
```

Comment:
I investigated this. ErrorCollector does indeed have a list of errors and a list of warnings. Unfortunately, it seems Message class was not designed in a way for me to extract text + line and column numbers. It only has a `write` method and implementers dump the information in plain text. So it seems that I cannot get structured output from an error collector.

refs:
- http://docs.groovy-lang.org/latest/html/gapi/org/codehaus/groovy/control/ErrorCollector.html
- http://docs.groovy-lang.org/latest/html/gapi/org/codehaus/groovy/control/messages/Message.html
- https://github.com/groovy/groovy-core/blob/master/src/main/org/codehaus/groovy/control/messages/Message.java